### PR TITLE
obs-ndi: add symlinks for plugin files

### LINKED ISSUE #161490: `obs-ndi` isn’t found by OBS
### Verification

- [X] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [ ] I have retried my command with `--force`.
- [ ] I ran `brew update-reset && brew update` and retried my command.
- [ ] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [X] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [X] I made doubly sure this is not a [checksum does not match / SHA256 mismatch](https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match) error (do not open an issue before trying to open a PR to fix first).

### Description of issue

Installing `obs-ndi` puts the `obs-ndi.plugin` in `/Library/Application Support/obs-studio/` instead of `~/Library/Application Support/obs-studio/`, and OBS doesn’t pick it up.

It’s strange to me, because downloading and running https://github.com/obs-ndi/obs-ndi/releases/download/4.13.0/obs-ndi-4.13.0-macos-universal.pkg by hand puts the files in the right directory 🤔

### Command that failed

No command failed.

### Output of command with `--verbose --debug`

```shell
I don’t think this applies either.
```

### Output of `brew doctor` and `brew config`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libndi.dylib


**Note that libndi, on which obs-ndi depends, is installing a dylib that Homebrew reports as unexpected. Despite this everything works well.**
```

### Output of `brew tap`

```shell
homebrew/cask
homebrew/cask-fonts
homebrew/cask-versions
homebrew/core
```

### PR DESCRIPTION
**Pull request by bevanjkay**
This is a potential fix for https://github.com/Homebrew/homebrew-cask/issues/161490 using symlinks.

The issue is that the `pkg` installers create the plugin files in `/Library/Application Support/` because they are run with `-target /`.
OBS Studio expects the files to reside in the user's directory `~/Library/Application Support/` so create symlinks to correctly link them.

This issue seems to affect ALL of the plugins in `homebrew-cask` that are for OBS Studio that use a `pkg` to install.
If this is an acceptable fix, then we should replicate this across the others.

CC: @carlocab - thanks for the suggestion on the issue from February - https://github.com/Homebrew/homebrew-cask/pull/139807#discussion_r1105679828

Closes #161490

### CODE REVIEW

**Review by razvanazamfirei — APPROVED** [review]

**Comment by razvanazamfirei** [comment]
This looks good to me!

**Review by krehel — APPROVED** [review]

**Comment by leafac** [comment]
I just installed the latest version including this pull request and everything works 👏

Thank you @bevanjkay for preparing the pull request, @razvanazamfirei & @krehel for reviewing, and @miccal for merging! ❤️

I’m curious: Could there be a way for a Cask to specify a custom `-target`, which perhaps would address this issue in a more principled way than symlinks?

**Comment by bevanjkay** [comment]
It's because in principle Homebrew attempts to install casks so that they are accessible system-wide and not just for the current logged in user. This works in almost every case, except for this one.

File: Casks/o/obs-ndi.rb [hunk 1]
```diff
@@ -16,6 +16,26 @@
...
+  # however OBS Studio expects them to be in ~/Library/Application Support/obs-studio/plugins
+  # so we create symlinks to correctly link the plugin files for OBS Studio.
+  postflight do
+    puts "Creating #{token} symlinks in ~/Library/Application Support/obs-studio/plugins"
+    target = Pathname.new("~/Library/Application Support/obs-studio/plugins").expand_path
+    source = "/Library/Application Support/obs-studio/plugins"
+
+    FileUtils.mkdir_p target
+    File.symlink("#{source}/obs-ndi.plugin", "#{target}/obs-ndi.plugin")
+    File.symlink("#{source}/obs-ndi.plugin.dSYM", "#{target}/obs-ndi.plugin.dSYM")
```
Comment (carlocab) [inline Casks/o/obs-ndi.rb:29]: Kinda weird that they include `.dSYM` (debug symbols) in their release assets.

File: Casks/o/obs-ndi.rb [hunk 2]
```diff
@@ -16,6 +16,26 @@
 
   pkg "obs-ndi-#{version}-macos-universal.pkg"
 
+  # The pkg installs the plugin files to /Library/Application Support/obs-studio/plugins
+  # however OBS Studio expects them to be in ~/Library/Application Support/obs-studio/plugins
+  # so we create symlinks to correctly link the plugin files for OBS Studio.
+  postflight do
```
Comment (carlocab) [inline Casks/o/obs-ndi.rb:22]: I think this won't work if the user who did `brew install obs-ndi` isn't the one who is actually using it. But poor multi-user support is true more generally about `brew` as well.

**Review by carlocab — COMMENTED** [review]
> CC: @carlocab - thanks for the suggestion on the issue from February - [#139807 (comment)](https://github.com/Homebrew/homebrew-cask/pull/139807#discussion_r1105679828)

This was actually @reitermarkus' suggestion.

File: Casks/o/obs-ndi.rb (hunk 2 above)
Comment (bevanjkay) [reply Casks/o/obs-ndi.rb:22]: That's the way the `pkg` file installs the standard way, it only lets you install to the current user.

Comment (bevanjkay) [reply Casks/o/obs-ndi.rb:22]: Maybe we should also add a caveat?

Comment (carlocab) [reply Casks/o/obs-ndi.rb:22]: > Maybe we should also add a caveat?

Maybe for OBS Studio, if at all. I'd leave it alone unless more users complain, though.

**Pull request merged**